geo: ZM prototype

otan opened the pull request:
Supporting ZM dimensions are blocked on WKT parsing (hard to test
otherwise!).

GEOS EWKB results do not return the 4th dimension unfortunately.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53089)
<!-- Reviewable:end -->

otan committed:
geo: ZM prototype

Supporting ZM dimensions are blocked on WKT parsing (hard to test
otherwise!).

Release note: None

otan commented:
not today!
